pkg/ccl/changefeedccl/helpers_test.go

```diff
@@ -899,17 +997,132 @@ func makeSpanGroupFromCheckpoint(
 	return spanGroup
 }
 
+var forceEnrichedEnvelope = metamorphic.ConstantWithTestBool("force-enriched-envelope", false)
+
+var forcedFeeds struct {
+	syncutil.Mutex
+	m map[jobspb.JobID]struct{}
+}
+
+func init() {
+	forcedFeeds.m = make(map[jobspb.JobID]struct{})
+}
+
+type optOutOfMetamorphicEnrichedEnvelope struct {
+	reason string
+}
+
 func feed(
 	t testing.TB, f cdctest.TestFeedFactory, create string, args ...interface{},
 ) cdctest.TestFeed {
 	t.Helper()
+
+	create, args, forced, err := maybeForceEnrichedEnvelope(t, create, f, args)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	feed, err := f.Feed(create, args...)
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	if forced {
+		jobID := feed.(cdctest.EnterpriseTestFeed).JobID()
+		func() {
+			forcedFeeds.Lock()
+			defer forcedFeeds.Unlock()
+			forcedFeeds.m[jobID] = struct{}{}
```

Comment:
This wouldn't add any changefeeds that explicitly specified `enriched` right? Instead of having a global map, could we store some state (maybe on the `jobFeed` struct) based on whether the changefeed has the `enriched` option and have some function in `cdctest.EnterpriseTestFeed` that returns whether the feed uses enriched?